src/server/pps/server/api_server.go

```diff
@@ -976,12 +976,13 @@ func (a *apiServer) stopJob(txnCtx *txncontext.TransactionContext, job *pps.Job,
 		}); err != nil && !pfsServer.IsCommitNotFoundErr(err) && !pfsServer.IsCommitDeletedErr(err) && !pfsServer.IsCommitFinishedErr(err) {
 			return err
 		}
-
-		if err := a.env.PfsServer().FinishCommitInTransaction(txnCtx, &pfs.FinishCommitRequest{
-			Commit: ppsutil.GetStatsCommit(commitInfo),
-			Empty:  true,
-		}); err != nil && !pfsServer.IsCommitNotFoundErr(err) && !pfsServer.IsCommitDeletedErr(err) && !pfsServer.IsCommitFinishedErr(err) {
-			return err
+		if statsCommit := ppsutil.GetStatsCommit(commitInfo); statsCommit != nil {
```

Comment:
this check is basically handles: `if commit_type != spout`. Is there a clearer way to express this?

Reply:
that's good to know. I think i'll leave it as a null check and just add a comment explaining the cases

Reply:
Like I mentioned above, I think we should follow the more standard convention of returning an error rather than a nil value. I do kind of prefer not actually calling that function when there is no expectation of a stats commit existing, so we could just put a panic in that function and hard fail when anything attempts to get a stats commit when it shouldn't.